aea/mail/base.py

```diff
@@ -242,6 +242,8 @@ def put_message(self, to: Optional[Address] = None, sender: Optional[Address] =
 class Connection:
     """Abstract definition of a connection."""
 
+    bridge: Optional[Any]
```

Comment:
This is an annoying requirement of mypy. It always checks against the base class and throws an error if an instance var is not referenced there.

Reply:
Removing that line, I get only the following error:
```
aea/channel/oef.py:526: error: "Connection" has no attribute "bridge"
```

That corresponds to the `return` statement of the following snippet:
```
    @property
    def mail_stats(self) -> MailStats:
        """Get the mail stats object."""
        return self._connection.bridge.mail_stats
```

Putting the usual `# type: ignore` will make the linter to skip the problem.

I'm wondering if this could be a better solution. The advantage would be that we remove the concept of `bridge` in the abstract `Connection` class. In fact, the `bridge` attribute in the `OEFConnection` is an "implementation detail", and doesn't correspond to an abstract concept. However, we could surely design and develop an abstract `class Bridge` in order to solve a more general problem in the connection implementation.

Reply:
Having looked at the code once more we should probably consider renaming `bridge` to `channel`. I then fixed all the channel logic across the 3 examples. It now is consistent. Let me know what you think.